### PR TITLE
[Fix] deepsparse – tokenizer stride size issue with original transformers

### PR DESCRIPTION
**Pull request by dbogunowicz**
Fix for: https://app.asana.com/0/1205724081986514/1205993362080974
(the exhaustive explanation for this fix is in the ticket)
@dbarbuzzi this fix works for me, please confirm whether it also solves your issue)
Will cherry-pick once approved.

### CODE REVIEW

**Comment by dbarbuzzi** [comment]
The changes work with the minimal example reported. Quick question, though, is it expected that if the consumer uses a non-default sequence_length that they should also be manually defining the doc_stride? I ask because now if I e.g. set a sequence_length of 32 in the constructor with no other kwargs, it will fail the new check in the PR as the doc_stride simply defaults to `62` regardless.

**Comment by dbogunowicz** [comment]
@dbarbuzzi I understand your concerns. Let me take a look at what are HF default settings.

**Comment by dbogunowicz** [comment]
@dbarbuzzi 
https://github.com/huggingface/transformers/blob/main/src/transformers/pipelines/question_answering.py#L408
It's pretty explicit here.

**Comment by dbogunowicz** [comment]
@dbarbuzzi let's set doc_stride to None by default and compute it dynamically to avoid the issue. I think this is fine.

File: src/deepsparse/transformers/pipelines/question_answering.py [hunk 1]
```diff
@@ -160,6 +160,16 @@ def __init__(
 
         super().__init__(**kwargs)
 
+        if doc_stride is None:
```
Comment (bfineran) [inline src/deepsparse/transformers/pipelines/question_answering.py:163]: let's see if we can read the number of special tokens from the tokenizer or tokenizer config and cap doc stride at `min(doc_stride, sequence_len - num_special_tokens)`

Comment (bfineran) [reply src/deepsparse/transformers/pipelines/question_answering.py:163]: and then let's add short warning

**Review by bfineran — CHANGES_REQUESTED** [review]

**Review by bfineran — APPROVED** [review]

**Pull request merged**